internal/controller/olsconfig_controller.go

```diff
@@ -141,7 +145,12 @@ func (r *OLSConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// Update status condition for Custom Resource
 	r.updateStatusCondition(ctx, olsconfig, typeCRReconciled, true, "Custom resource successfully reconciled", nil)
 
-	return ctrl.Result{}, nil
+	// Requeue if no reconciliation is scheduled in future.
+	if r.NextReconcileTime.After(time.Now()) {
+		return ctrl.Result{}, nil
+	}
+	r.NextReconcileTime = time.Now().Add(time.Duration(r.Options.ReconcileInterval) * time.Second)
+	return ctrl.Result{RequeueAfter: time.Duration(r.Options.ReconcileInterval) * time.Second}, nil
```

Comment:
Why is this `RequeueAfter` value different from the ones that are above?

Reply:
In case of error, the operator should retry it rapidly, attempting to get recovered from temporary hiccups issues.
When no error happens, it is a normal periodic reconciliation, to catch up any event may miss out in last reconciliation cycle.